mediapipe/graphs/instantmotiontracking/calculators/matrices_manager_calculator.cc

```diff
@@ -0,0 +1,324 @@
+  const float yaw = cc->Inputs().Tag(kIMUDataTag).Get<float[]>()[0];
+  const float pitch = cc->Inputs().Tag(kIMUDataTag).Get<float[]>()[1];
+  const float roll = cc->Inputs().Tag(kIMUDataTag).Get<float[]>()[2];
+  // We can pre-generate the IMU submatrix as this remains unchanged for all objects
+  const Matrix3f imu_rotation_submatrix = GenerateIMURotationSubmatrix(yaw, pitch, roll);
+
+  int render_idx = 0;
+  for (const Anchor &anchor : anchor_data) {
+    const int id = anchor.sticker_id;
+
+    TimedModelMatrixProto* model_matrix =
+        model_matrix_list->add_model_matrix();
+    model_matrix->set_id(id);
+
+    // The user transformation data associate with this sticker must be defined
+    const float rotation = GetUserRotation(user_rotation_data, id);
+    const float scaler = GetUserScaler(user_scaling_data, id);
+
+    // A vector representative of a user's sticker rotation transformation can be created
+    const Matrix3f user_rotation_submatrix = GenerateUserRotationMatrix(rotation);
+    // Next, the submatrix representative of the user's scaling transformation must be generated
+    const DiagonalMatrix3f user_scaling_submatrix = GenerateScalingMatrix(scaler);
+
+    // The user transformation data can be concatenated into a final rotation submatrix with the
+    // device IMU rotational data
+    const Matrix3f user_transformed_rotation_submatrix = user_scaling_submatrix * imu_rotation_submatrix * user_rotation_submatrix;
+
+    // We can also generate the initial scaling diagonal based on the preset scale factor for this render id
+    const DiagonalMatrix3f render_scaling_diagonal = GenerateScalingMatrix(GetDefaultRenderScale(render_data[render_idx++]));
+
+    // A vector representative of the translation of the object in OpenGL coordinate space must be generated
+    const Vector3f translation_vector = GenerateAnchorVector(anchor);
+
+    // Concatenate all model matrix data
+    const Matrix4fCM final_model_matrix = GenerateEigenModelMatrix(translation_vector, render_scaling_diagonal * user_transformed_rotation_submatrix);
+
+    // The generated model matrix must be mapped to TimedModelMatrixProto (col-wise)
+    for (int x = 0; x < final_model_matrix.rows(); ++x) {
+      for (int y = 0; y < final_model_matrix.cols(); ++y) {
+        model_matrix->add_matrix_entries(final_model_matrix(x, y));
+      }
+    }
+  }
+
+  cc->Outputs()
+      .Tag(kModelMatricesTag)
+      .Add(model_matrices.release(), cc->InputTimestamp());
+
+  return ::mediapipe::OkStatus();
+}
+
+// Using a specified rotation value in radians, generate a rotation matrix for use with base rotation submatrix
+const Matrix3f MatricesManagerCalculator::GenerateUserRotationMatrix(const float rotation_radians) {
+  Eigen::Matrix3f user_rotation_submatrix;
+    user_rotation_submatrix =
+        // The rotation in radians must be inverted to rotate the object
+        // with the direction of finger movement from the user (system dependent)
```

Comment:
super-nit: so that the calculator itself is x-platform / system independent, I'd just say something like "we assume rotation angles are passed in [clockwise/counterclockwise], so we must negate the rotation in order to rotate the object to match", or something like that.  That way it's up to the responsibility of the caller (regardless of system/platform) to call appropriately in the future.